src/apps/monitoring/iplabel/newtest/restapi/mode/scenarios.pm

```diff
@@ -130,10 +131,14 @@ sub manage_selection {
         (defined($self->{option_results}->{filter_scenario_name}) ? md5_hex($self->{option_results}->{filter_scenario_name}) : md5_hex('all'));
     my $last_timestamp = $self->read_statefile_key(key => 'last_timestamp');
     my $timespan = 5;
+    if (defined($self->{option_results}->{timeframe}) && $self->{option_results}->{timeframe} =~ /^(\d+)$/) {
+        $timespan = $1 / 60;
+    }
+
     if (defined($last_timestamp)) {
         $timespan = POSIX::ceil((time() - $last_timestamp) / 60);
     } else {
-        $last_timestamp = time() - (60 * 5);
+        $last_timestamp = time() - (60 * $timespan);
```

Comment:
At this time the code checks at each execution if the timeframe option is defined by user even if 
$last_timestamp can be calculated from cache file. It may be more relevent to avoid checking the 
values if not needed.
After reading ticket, it remain unclear for me if the user wants to set timeframe option for first 
execution only or for all executions. I suggest both cases below. And also I tried to uniformize 
times in second for $timespan definition, removing the POSIX::ceil call too.

First case : First range is calculated with default value or with timeframe option set by user
then it's calculated from the timestamp present in the cache file.

```
    my $last_timestamp = $self->read_statefile_key(key => 'last_timestamp');

    my $timespan = 300; # default value: 5 min (only use if not set by user at the first time)

    if (defined($last_timestamp)) {
        $timespan = time() - $last_timestamp;
    }else{
    	if (defined($self->{option_results}->{timeframe}) && $self->{option_results}->{timeframe} =~ /^(\d+)$/) {
             $timespan = $1;
        }
        $last_timestamp = time() - $timespan;
    }
```

Second case : User wants to use at every execution the same range set in timeframe.
If not, first range is calculated with default value and then it's calculated from 
the timestamp present in the cache file.

```
    my $last_timestamp = $self->read_statefile_key(key => 'last_timestamp');

    my $timespan = 300; # default value: 5 min (only use if not set by user at the first time)

    if (defined($self->{option_results}->{timeframe}) && $self->{option_results}->{timeframe} =~ /^(\d+)$/) {
        $timespan = $1;
        $last_timestamp = time() - $timespan;
    }else{
        if (defined($last_timestamp)) {
             $timespan = time() - $last_timestamp;
        } else {
             $last_timestamp = time() - $timespan;
        }
    }
```